persist/encoding/decoder.go

```diff
@@ -42,3 +51,110 @@ type Decoder interface {
 	// DecodeLogEntry decodes commit log entry
 	DecodeLogEntry() (schema.LogEntry, error)
 }
+
+// DecoderStream is a data stream that is read by the decoder,
```

Comment:
This comment is a little confusing. After reading it, I expected the constructor for DecoderStream to look like:

```golang
NewDecoderStream(b []byte, r io.Reader) DecoderStream {}
```

but instead it only accepts []byte, and the type of the internal reader is *bytes.Reader which is not an interface.

How would you use this interface with a ReaderWithDigest interface?

Reply:
`NewDecoderStream(...)` in this package is just a default offering for users to use if they don't need to provide the reader themselves.

There is a private `newDecoderStream(...)` in the `persist/fs package` for that use case.  This is so that users who don't need to and already have just a byte slice can still easily pass a buffer to decode into the msgpack decoder.